code/go/0chain.net/core/util/columnizer.go

```diff
@@ -0,0 +1,61 @@
+package util
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+var (
+	firstCapRegex 	= regexp.MustCompile("(.)([A-Z][a-z]+)")
+	allCapRegex 	= regexp.MustCompile("([a-z0-9])([A-Z])")
+	gormColumRegex 	= regexp.MustCompile("(?m)column:([a-zA-Z0-9_]+);")
+)
+
+
+func toSnakeCase(str string) string {
+    snake := firstCapRegex.ReplaceAllString(str, "${1}_${2}")
+    snake = allCapRegex.ReplaceAllString(snake, "${1}_${2}")
+    return strings.ToLower(snake)
+}
+
+func Columnize[T any](objects []T) (map[string][]interface{}, error) {
+	columns := make(map[string][]interface{})
+	
+	for _, obj := range objects {
+		v := reflect.ValueOf(obj)
+
+		if v.Kind() == reflect.Ptr {
+			v = v.Elem()
+		}
+
+		if v.Kind() != reflect.Struct {
+			return nil, fmt.Errorf("columnize error: type invalid")
+		}
+
+		t := v.Type()
+
+		for fidx := 0; fidx < t.NumField(); fidx++ {
+			f := t.Field(fidx)
+			
+			// Resolve field key in this order: gorm.column Tag > snake-casing FieldName
```

Comment:
We need tests to ensure the field keys are resolved as expected. So let's add tests to cover all the cases.